rts/Sim/MoveTypes/HoverAirMoveType.cpp

```diff
@@ -452,19 +452,13 @@ void CHoverAirMoveType::UpdateFlying()
 				// NOTE: should CMD_LOAD_ONTO be here?
 				const bool isTransporter = owner->unitDef->IsTransportUnit();
 				const bool hasLoadCmds = isTransporter && (topCmdID == CMD_LOAD_ONTO || topCmdID == CMD_LOAD_UNITS);
-				// [?] transport aircraft need some time to detect that they can pickup
-				const bool canLoad = isTransporter && (++waitCounter < ((GAME_SPEED << 1) - 5));
 				const bool isBusy = IsUnitBusy(owner);
 
-				if (!CanLand(isBusy) || (canLoad && hasLoadCmds)) {
+				if (!CanLand(isBusy) || hasLoadCmds) {
 					wantedSpeed = ZeroVector;
 
 					if (isTransporter) {
-						if (waitCounter > (GAME_SPEED << 1)) {
-							wantedHeight = orgWantedHeight;
-						}
-
-						SetState(AIRCRAFT_HOVERING);
```

Comment:
It occured to me that for hovering to make sense here, the hovering has to guarantee that it will cross goalPos soon. And as the hovering is random, it can't.